src/Raven.Server/Documents/Handlers/TransformerHandler.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Raven.Abstractions.Indexing;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class TransformerHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/transformers", "PUT")]
+        public async Task Put()
+        {
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var json = await context.ReadForDiskAsync(RequestBodyStream(), name);
```

Comment:
Can use ToMemory here, we don't need to compress the data, we are going to be saving it as full in a bit anyway